diff --git a/node_modules/shikiji-core/dist/index.mjs b/node_modules/shikiji-core/dist/index.mjs
index 971ac9e..66e3443 100644
--- a/node_modules/shikiji-core/dist/index.mjs
+++ b/node_modules/shikiji-core/dist/index.mjs
@@ -4866,6 +4866,7 @@ async function getShikiInternal(options = {}) {
     const _registry = new Registry(resolver, themes, langs);
     Object.assign(_registry.alias, options.langAlias);
     await _registry.init();
+    let _currentTheme;
     function getLangGrammar(name) {
         const _lang = _registry.getGrammar(name);
         if (!_lang)
@@ -4880,7 +4881,10 @@ async function getShikiInternal(options = {}) {
     }
     function setTheme(name) {
         const theme = getTheme(name);
-        _registry.setTheme(theme);
+        if (_currentTheme !== name) {
+          _registry.setTheme(theme);
+          _currentTheme = name;
+        }
         const colorMap = _registry.getColorMap();
         return {
             theme,
